core/src/main/scala/topgun/core/CallSite.scala

```diff
@@ -20,14 +20,15 @@ object CallSite {
   }
   private val USER_FRAME = 1
   private val IGNORABLE_TOP_FRAME = 2
+  val loaderMap = mutable.HashMap.empty[Int, ClassLoader]
 
   private val all = new mutable.HashMap[CallSite, CallSite]()
-  def apply(packageName: String, className: String, methodName: String, desc: String, line: Int): CallSite = {
-    val site = new CallSite(packageName.intern, className.intern, methodName.intern, desc.intern, line)
+  def apply(packageName: String, className: String, methodName: String, desc: String, line: Int, classLoaderKey: Int): CallSite = {
```

Comment:
so if `classloaderKey: Int` becomes `classLoader: ClassLoader`
you can add 
lazy val classInfo = ...

(from the other PR, when its ready)

Reply:
we need to pass information about the ClassLoader to the individual CallSites. if "classLoader: ClassLoader" becomes the key, then we will be sending classLoader objects to all the callsites. and also if classloader is used as key, hashmap will still extract the hash value of the classloader which is actually include the classpaths

Reply:
you need "information about" the lassloader, so you can get to the classloader

2 JFR files that have the same classpath refer to the same classloader, classes, asm values etc
2 strings with the same hashcode may or may not be the same

we need to pass the accessor for asm to do it work, so the objects that holds the classloader and the Map/cache of ClassInfo

Reply:
If I understand this well, I willl change classloaderKey: Int  to classLoader: ClassLoader, pass the classLoader: ClassLoader object to the individual callsites and when we need classInfo we use the classLoader object. Am I right?

Reply:
yes
https://github.com/mkeskells/TopGun/pull/7 adds a wrapper for the classloader and the cache, but unless that one goes in first , you just need the classloader